### PR TITLE
Populate container.id field in container metricset

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
## What does this PR do?

This PR adds `container.id` and `container.runtime` fields in events generated by kubernetes `container` Metricset.

## Why is it important?

`Container.id` is a unique identifier for a Kubernetes container and is very important to be part in all kubernetes events.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- 

## Screenshots

container id

### CODE REVIEW

File: metricbeat/module/kubernetes/types.go [hunk 1]
```diff
@@ -134,3 +136,6 @@ type Summary struct {
 		} `json:"volume"`
 	} `json:"pods"`
 }
+
+// PodContainerStatus data
+type PodContainerStatus = v1.ContainerStatus
```
Comment (ChrsMark) [inline metricbeat/module/kubernetes/types.go:141]: Why we need to declare it again? Can't we re-use the already existent https://github.com/elastic/beats/blob/cf1c78c85dda1fd0c3d2e072efd52b903da2cdb0/libbeat/common/kubernetes/types.go#L62-L61 ?

File: metricbeat/module/kubernetes/container/container.go [hunk 2]
```diff
@@ -106,11 +108,35 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 
 	for _, event := range events {
 
+		// applying ECS to kubernetes.container.id in the form <container.runtime>://<container.id>
+		// copy to ECS fields the kubernetes.container.image, kubernetes.container.name
+		containerFields := common.MapStr{}
+		if containerID, ok := event["id"]; ok {
+			cID := (containerID).(string)
```
Comment (ChrsMark) [inline metricbeat/module/kubernetes/container/container.go:115]: nit: This block could be a helper function.

File: metricbeat/module/kubernetes/util/kubernetes.go [hunk 3]
```diff
@@ -215,7 +223,11 @@ func NewContainerMetadataEnricher(
 
 				id := join(pod.GetObjectMeta().GetNamespace(), pod.GetObjectMeta().GetName(), container.Name)
 				m[id] = meta
+				if s, ok := statuses[container.Name]; ok {
+					PerfMetrics.ContainerID.SetStringVal(cuid, s.ContainerID)
+				}
```
Comment (ChrsMark) [inline metricbeat/module/kubernetes/util/kubernetes.go:228]: Wondering if we could avoid using the cache for this and use `meta` directly.

**Review by ChrsMark — COMMENTED** [review]
Really useful addition! Left some comments.

File: metricbeat/module/kubernetes/util/kubernetes.go (hunk 3 above)
Comment (MichaelKatsoulis) [reply metricbeat/module/kubernetes/util/kubernetes.go:230]: Yes I will do this! Good catch

File: metricbeat/module/kubernetes/types.go (hunk 1 above)
Comment (MichaelKatsoulis) [reply metricbeat/module/kubernetes/types.go:141]: You are right. It is not needed

**Comment by MichaelKatsoulis** [comment]
@ChrsMark I added the container.id and container.runtime in the meta fields

**Review by ChrsMark — APPROVED** [review]
🚀

File: metricbeat/module/kubernetes/util/kubernetes.go [hunk 4]
```diff
@@ -213,6 +221,13 @@ func NewContainerMetadataEnricher(
 					}
 				}
 
+				if s, ok := statuses[container.Name]; ok {
+					split := strings.Index(s.ContainerID, "://")
+					if split != -1 {
+						meta.Put("container.id", s.ContainerID[split+3:])
```
Comment (tetianakravchenko) [inline metricbeat/module/kubernetes/util/kubernetes.go:227]: could you please add a comment with ab example of `statuses[container.Name]` value, that could explain why those offsets are used

Comment (MichaelKatsoulis) [reply metricbeat/module/kubernetes/util/kubernetes.go:229]: I added a comment in https://github.com/elastic/beats/pull/29560/commits/755b1f23cd0eb218964b736e3250499f87fc55cc

**Review by tetianakravchenko — APPROVED** [review]

**Pull request merged**